enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/scenarios/ReadReplicaToReadReplicaCatchupIT.java

```diff
@@ -0,0 +1,268 @@
+            tx.success();
+        } );
+
+        storeSomeDataInCores( cluster, numberOfNodes );
+
+        AtomicBoolean labelScanStoreCorrectlyPlaced = new AtomicBoolean( false );
+        Monitors monitors = new Monitors();
+
+        ReadReplica firstReadReplica =
+                cluster.addReadReplicaWithIdAndMonitors( firstReadReplicaLocalMemberId, monitors );
+        File labelScanStore = LabelScanStoreProvider
+                .getStoreDirectory( new File( firstReadReplica.storeDir(), TEMP_COPY_DIRECTORY_NAME ) );
+
+        monitors.addMonitorListener( (FileCopyMonitor) file ->
+        {
+            if ( file.getParent().contains( labelScanStore.getPath() ) )
+            {
+                labelScanStoreCorrectlyPlaced.set( true );
+            }
+        } );
+
+        firstReadReplica.start();
+
+        checkDataHasReplicatedToReadReplicas( cluster, numberOfNodes );
+
+        upstreamFactory.setCurrent( firstReadReplica );
+
+        ReadReplica secondReadReplica = cluster.addReadReplicaWithId( 202 );
+        secondReadReplica.setUpstreamDatabaseSelectionStrategy( "specific" );
+
+        secondReadReplica.start();
+
+        checkDataHasReplicatedToReadReplicas( cluster, numberOfNodes );
+
+        firstReadReplica.shutdown();
+        upstreamFactory.reset();
+
+        cluster.removeReadReplicaWithMemberId( firstReadReplicaLocalMemberId );
+
+        // when
+        // More transactions into core
+        storeSomeDataInCores( cluster, numberOfNodes );
+
+        // then
+        // reached second read replica from cores
+        checkDataHasReplicatedToReadReplicas( cluster, numberOfNodes * 2 );
+    }
+
+    private void storeSomeDataInCores( Cluster cluster, int numberOfNodes ) throws Exception
+    {
+        for ( int i = 0; i < numberOfNodes; i++ )
+        {
+            cluster.coreTx( ( db, tx ) ->
+            {
+                createData( db );
+                tx.success();
+            } );
+        }
+    }
+
+    private void createData( GraphDatabaseService db )
+    {
+        Node node = db.createNode( Label.label( "Foo" ) );
+        node.setProperty( "foobar", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar1", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar2", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar3", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar4", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar5", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar6", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar7", format( "baz_bat%s", UUID.randomUUID() ) );
+        node.setProperty( "foobar8", format( "baz_bat%s", UUID.randomUUID() ) );
+    }
+
+    private void checkDataHasReplicatedToReadReplicas( Cluster cluster, long numberOfNodes ) throws Exception
```

Comment:
We already have helpers which check that two different members have the exact same data. See Cluster#dataMatchesEventually.